src/algebra/parity.lean

```diff
@@ -85,6 +86,27 @@ begin
   refine ⟨m * n, mul_mul_mul_comm m m n n⟩,
 end
 
+section monoid
+
+@[simp]
+lemma is_square_zero [mul_zero_class α] : is_square (0 : α) := ⟨0, (mul_zero _).symm⟩
+
+@[simp]
+lemma irreducible.not_square [comm_monoid α] {x : α} (h : irreducible x) :
```

Comment:
Can you also add `is_square.not_irreducible`?

Reply:
Done!